troposphere/static/js/components/modals/allocationSource/NoAllocationSourceModal.react.js

```diff
@@ -0,0 +1,310 @@
+        projects: React.PropTypes.instanceOf(Backbone.Collection).isRequired,
+        allocationSources: React.PropTypes.instanceOf(Backbone.Collection).isRequired,
+        onConfirm: React.PropTypes.func.isRequired
+    },
+
+    getInitialState() {
+        return this.getStateFromProps(this.props);
+    },
+
+    componentWillReceiveProps(props) {
+        this.setState(this.getStateFromProps(props));
+    },
+
+    getStateFromProps(props) {
+        let { allocationSources, instances } = this.props;
+
+        // This is a structure for the pairing of (instance, allocationSource)
+        // Ex. { instance.id:  { instance, allocationSource }}
+        let instanceAllocations = {};
+        let defaultAllocation = allocationSources.first();
+
+        // Create a default pairing for each instance
+        instances.forEach(instance => {
+            instanceAllocations[instance.id] = {
+                allocationSource: defaultAllocation,
+                instance
+            }
+        })
+
+        return {
+            instanceAllocations,
+        }
+    },
+
+    onConfirm() {
+        // instanceAllocations takes the form of:
+        // {
+        //     instanceId: { allocationSource, instance }
+        // }
+        let { instanceAllocations } = this.state;
+
+        // Flatten it to:
+        // [ { allocationSource, instance } ]
+        let flattened = _.values(instanceAllocations);
+
+        this.props.onConfirm(flattened);
+    },
+
+    renderInstance(instance) {
+        let { allocationSources } = this.props;
+        let { allocationSource } = this.state.instanceAllocations[instance.id];
+
+        return (
+        <div
+            key={ instance.id }
+            style={{
+                display: "flex",
+                alignItems: "center",
+                justifyContent: "space-between",
+                padding: "0px 0px 10px 10px",
+            }} >
+            <b style={{whiteSpace: "nowrap"}}>
+                { instance.get('name') }
+            </b>
+            <span style={{width: "40%", float: "right"}}>
+                <SelectMenu
+                    current={ allocationSource }
+                    list={ allocationSources }
+                    onSelect={ as => this.pairInstanceWithAllocation(instance, as) }
+                    optionName={ as => as.get("name") }
+                    />
+            </span>
+        </div>
+        )
+    },
+
+    pairInstanceWithAllocation(instance, allocationSource) {
+        let { instanceAllocations } = this.state;
+        instanceAllocations[instance.id] = {
+            instance,
+            allocationSource,
+        }
+        this.setState({ instanceAllocations });
+    },
+
+    renderProject(project, orphans) {
+        let name = project.get('name');
+        let renderedInstances = orphans.map(this.renderInstance);
+
+        return (
+            <div key={ name }>
+                <h2 className="t-title">{ `Project: ${name}` }</h2>
```

Comment:
I probably did this but we should aim to put tags above and below children. This way we can scan indentation to see blocks and confirm that they are closed.